spec/validators/date_in_range_validator_spec.rb

```diff
@@ -0,0 +1,47 @@
+require "rails_helper"
+
+RSpec.describe DateInRangeValidator do
+  subject do
```

Comment:
Good point; I started out with an abstract class because *in theory* a validator should be able to validate something without knowing the class name. I had to add the class name otherwise the error string wouldn't be created properly. I could have used a Factory in that case - either a Transaction or an Activity (which would require the attribute name to be slightly different). I'm not fussed either way - am happy to make it a Factory.

Reply:
Actually, scratch that. I've attempted to use a Transaction factory, but on that model the date is mandatory, so testing the validator with a nil date returns `false` because the model requires a date. 

I think I'd rather leave it with an anonymous class because that ensures we're only testing the validator, not the class it's applied to. If that's OK?